platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+            return Response.ok(redirectUpdated).build();
+        }
+        throw new UnsupportedOperationException("Must use HTTP POST or Redirect bindings.");
+    }
+
+    protected AuthnRequest unwrapAuthnRequest(String samlRequest, boolean isPost) {
+        LOGGER.debug("Creating AuthnRequest object from SAMLRequest string.");
+        if (StringUtils.isEmpty(samlRequest)) {
+            throw new IllegalArgumentException("Missing SAMLRequest on IdP request.");
+        }
+        String decodedRequest;
+        if (isPost) {
+            decodedRequest = new String(Base64.decodeBase64(samlRequest));
+        } else {
+            try {
+                decodedRequest = RestSecurity.inflateBase64(samlRequest);
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Unable to decode SAMLRequest: base64/inflate.");
+            }
+        }
+        ByteArrayInputStream tokenStream = new ByteArrayInputStream(decodedRequest.getBytes());
+        Document authnDoc;
+        try {
+            authnDoc = StaxUtils.read(new InputStreamReader(tokenStream, "UTF-8"));
+        } catch (Exception ex) {
+            throw new IllegalArgumentException("Unable to read SAMLRequest as XML.");
+        }
+        XMLObject authnXmlObj;
+        try {
+            authnXmlObj = OpenSAMLUtil.fromDom(authnDoc.getDocumentElement());
+        } catch (WSSecurityException ex) {
+            throw new IllegalArgumentException(
+                    "Unable to convert AuthnRequest document to XMLObject.");
+        }
+        if (!(authnXmlObj instanceof AuthnRequest)) {
+            throw new IllegalArgumentException("SAMLRequest object is not AuthnRequest.");
+        }
+        LOGGER.debug("Created AuthnRequest object successfully.");
+        return (AuthnRequest) authnXmlObj;
+    }
+
+    protected void validateAuthnRequest(AuthnRequest authnRequest, String samlRequest,
+            String relayState, String signatureAlgorithm, String signature, boolean isPost)
+            throws SimpleSign.SignatureException, ValidationException {
+        LOGGER.debug("Validating AuthnRequest required attributes and signature");
+        if (isPost) {
+            if (strictSignature) {
+                if (authnRequest.getSignature() != null) {
+                    SimpleSign.validateSignature(authnRequest.getSignature(),
+                            authnRequest.getDOM().getOwnerDocument(), getSignatureCrypto());
+                } else {
+                    throw new SimpleSign.SignatureException("No signature present on AuthnRequest.");
+                }
+            }
+        } else {
+            if (strictSignature) {
+                if (!StringUtils.isEmpty(signature) && !StringUtils.isEmpty(signatureAlgorithm)) {
+                    String signedParts;
+                    try {
+                        signedParts = SSOConstants.SAML_REQUEST + "=" + URLEncoder
```

Comment:
So why String.format instead of concatenation. Afaik, format is only good for localization (which I'm not doing) and concatenation is actually faster than format. Is there something else I'm missing?